source/common/router/config_impl.h

```diff
@@ -547,6 +547,27 @@ class InternalRedirectPolicyImpl : public InternalRedirectPolicy {
 };
 using DefaultInternalRedirectPolicy = ConstSingleton<InternalRedirectPolicyImpl>;
 
+/**
+ * Container for route config elements that pertain to a redirect.
+ *
+ * We keep them in a separate data structure to avoid memory overhead for the routes that do not use
+ * redirect.
+ */
+struct RedirectConfig {
```

Comment:
Comparing this to the `RedirectAction` message, it seems that [`response_code`](https://github.com/envoyproxy/envoy/blob/569ef492ba2ad1c9395a32fc3d143bb4501bc229/api/envoy/config/route/v3/route_components.proto#L1718). I'm not sure if it's worth moving it here or not, but just wondering what's the reason for not adding it here?

Reply:
There is no dedicated member variable for this. It is tracked in `direct_response_code_`, by `parseDirectResponseCode`, and can be either `redirect.response_code` or `direct_response.status`